state/state.go

```diff
@@ -27,3 +33,38 @@ type View interface {
 	NewView(ctx context.Context, changes merkledb.ViewChanges) (merkledb.View, error)
 	GetMerkleRoot(ctx context.Context) (ids.ID, error)
 }
+
+// StateKey holds the name of the key and its permission (Read/Write)
+// Specifying RWrite is setting both the Read and Write bit
+type Key struct {
+	Name string
+	// TODO: consider a dynamically sized []byte
+	Permission Permission
```

Comment:
You may want to comment that we use bits here instead of a set because we expect transactions to include this encoding in transactions.